torchio/transforms/augmentation/intensity/random_simulation.py

```diff
@@ -0,0 +1,172 @@
+                        else:
+                            data = image_dict[DATA]
+                        tissue = self.generate_tissue(data, mean, std)
+                        if final_image is None:
+                            final_image = Image(type=INTENSITY, affine=image_dict[AFFINE], tensor=tissue[0])
+                        else:
+                            final_image[DATA] += tissue
+
+        if original_image is not None:
+            final_image[DATA][final_image[DATA] == 0] = original_image[DATA][final_image[DATA] == 0]
+
+        sample[self.image_key] = final_image
+        sample.add_transform(self, random_parameters_images_dict)
+        return sample
+
+    def parse_coefficients(self, coefficients):
+        parsed_coefficients = {'mean': MEAN_RANGE, 'std': STD_RANGE}
+
+        if coefficients is None:
+            coefficients = {}
+
+        for label_key, dictionary in coefficients.items():
+            if list(dictionary.keys()) != ['mean', 'std']:
+                raise KeyError(f'Given coefficients {dictionary.keys()} do not match {["mean", "std"]}')
+            mean = self.parse_coefficient(dictionary['mean'], 'mean')
+            std = self.parse_coefficient(dictionary['std'], 'std')
+            parsed_coefficients.update({label_key: {'mean': mean, 'std': std}})
+
+        return parsed_coefficients
+
+    @staticmethod
+    def parse_coefficient(
+            nums_range: TypeRangeFloat,
+            name: str,
+            ) -> Tuple[float, float]:
+        if isinstance(nums_range, (int, float)):
+            return nums_range, nums_range
+
+        if len(nums_range) != 2:
+            raise ValueError(
+                f'If {name} is a sequence,'
+                f' it must be of len 2, not {nums_range}')
+        min_value, max_value = nums_range
+        if min_value > max_value:
+            raise ValueError(
+                f'If {name} is a sequence, the second value must be'
+                f' equal or greater than the first, not {nums_range}')
+        return min_value, max_value
+
+    def get_params(
+        self,
+        label: Union[str, TypeNumber]
+    ) -> Tuple[float, float]:
+        if label in self.coefficients:
+            mean_range, std_range = self.coefficients[label]['mean'], self.coefficients[label]['std']
+        else:
+            mean_range, std_range = self.coefficients['mean'], self.coefficients['std']
+
+        mean = torch.FloatTensor(1).uniform_(*mean_range).item()
+        std = torch.FloatTensor(1).uniform_(*std_range).item()
+        return mean, std
+
+    @staticmethod
+    def generate_tissue(
+            data: TypeData,
+            mean: TypeNumber,
+            std: TypeNumber,
+            ) -> TypeData:
+        # Create the simulated tissue using a gaussian random variable
+        data_shape = data.shape
+        gaussian = torch.randn(data_shape) * std + mean
```

Comment:
You could either use torch.randn_like or, even better, pass the shape instead of the data.

Reply:
I still need to use `data` to mask the generated volume.